pkg/ocm/sca/architectures_gather.go

```diff
@@ -0,0 +1,23 @@
+package sca
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// gatherArchitectures connects to K8S API to retrieve the list of
+// nodes and create a set of the present architectures
+func (c *Controller) gatherArchitectures(ctx context.Context) (map[string]struct{}, error) {
```

Comment:
it might be best to request all arches. MachineSets could be on an autoscaler when the gather action is executed.

Reply:
I don't get your point. This method is requesting all archs from the nodes list returned by k8s API. Could you please elaborate more?

Reply:
Right - I see that. Since you can scale machinesets from zero, you may be temporarily excluding some arches from your retrieval.

Reply:
Example https://issues.redhat.com/browse/OCPBUGS-27509 and these would be the second set of MachineSets created.

Reply:
I get your point now. If at the moment of retrieving the nodes list some machineset is scaled to 0, how can I get the node architecture? I think we can only gather the arch for the current existing nodes, not from those that are unavailable because any reason, but maybe there is another way to retrieve the nodes list...